File: dr_quine/PHP/Colleen.php
#!/usr/bin/php
<?php

/*
	one comment, one!
*/

function print_src($str)
{
	printf($str, 10, 39, $str);
}

function main()
{
	/*
		A second comment, for real?
	*/
	$str = '#!/usr/bin/php%1$c<?php%1$c%1$c/*%1$c	one comment, one!%1$c*/%1$c%1$cfunction print_src($str)%1$c{%1$c	printf($str, 10, 39, $str);%1$c}%1$c%1$cfunction main()%1$c{%1$c	/*%1$c		A second comment, for real?%1$c	*/%1$c	$str = %2$c%3$s%2$c;%1$c	print_src($str);%1$c}%1$c%1$cmain();%1$c';
	print_src($str);
}

main();

File: dr_quine/C/Colleen.c
#include <stdio.h>

/*
	One comment, one !
*/

void print_src (char *str)
{
	printf(str, 10, 9, str, 34);
}

int main (void)
{
	/*
		A second comment, for real ?
	*/
	char *str = "#include <stdio.h>%1$c%1$c/*%1$c%2$cOne comment, one !%1$c*/%1$c%1$cvoid print_src (char *str)%1$c{%1$c%2$cprintf(str, 10, 9, str, 34);%1$c}%1$c%1$cint main (void)%1$c{%1$c%2$c/*%1$c%2$c%2$cA second comment, for real ?%1$c%2$c*/%1$c%2$cchar *str = %4$c%3$s%4$c;%1$c%1$c%2$cprint_src(str);%1$c%1$c%2$creturn (0);%1$c}%1$c";

	print_src(str);

	return (0);
}
